main.go

```diff
@@ -14,11 +14,13 @@ import (
 var listenURI string
 var cloudURI string
 var apiKey string
+var applicationID string
 
 func main() {
 	flag.StringVar(&listenURI, "listen-uri", "0.0.0.0:8181", "Local TCP server address")
 	flag.StringVar(&cloudURI, "cloud-uri", "", "Cloud URI that edge tunneling service is running on")
 	flag.StringVar(&apiKey, "api-key", "", "API Key that needs to connect to edge tunneling service")
+	flag.StringVar(&applicationID, "application-id", "", "If JWT is used instead of api-key, application-id is needed")
```

Comment:
Is application id a required parameter? Should we check whether it is empty?

Reply:
This might go away if the application id is embedded in the jwt anyway, but more of architecture discussion. @zhucebuliaolongchuan what do you think?

Reply:
Yeah this won't be needed if application-id could be passed through already. But I am thinking about that if we could let the customers to use this client to sign their claims as well so they don't need to do that themselves. 

So if we want to chase that way, we need to have two options for them. First option is they could provide the key directly(they sign the claims themselves), or they could provide their private key only and then we could generate the jwt within this client.

Any ideas? @sk-arm @brianesquilona

Reply:
Hi Brian, until they make that architecture change we should go with this approach. We should track that architecture change separately and adjust the code accordingly when it happens. Otherwise we will be in waiting mode. 

Hi Chuan, is there an item in the JIRA to track that architecture change?

Chuan, regarding signing the JWT using their private key,  I think the key idea is customer never need to share their private key (they may have a secure store for their private key) with us (our tools, even if it is in their local machine) and that option may not acceptable. We can certainly ask PM.

Reply:
The architecture change is already in place, while it doesn't affect the client side really since `x-application-id` is a header being passed after the token is verified and it only happens between IAM service and our cloud service(fog-proxy-cloud), so I don't think we should worry about this here.

Yeah. This sounds more like a Product decision.

Reply:
@zhucebuliaolongchuan what is the conclusion? If we we plan to keep the application-id as command line we should go ahead with this change.

Reply:
@sk-arm It seems like we need to provide this. Maybe we could name it `application-id`? As the user, they don't need to know the name of the header. IMO `application-id` is more user friendly.